public/app/plugins/datasource/prometheus/querybuilder/shared/operationUtils.ts

```diff
@@ -23,6 +23,65 @@ export function functionRendererRight(model: QueryBuilderOperation, def: QueryBu
   return str + params.join(', ') + ')';
 }
 
+function rangeRendererWithParams(
+  model: QueryBuilderOperation,
+  def: QueryBuilderOperationDef,
+  innerExpr: string,
+  renderLeft: boolean
+) {
+  if (def.params.length < 2) {
+    throw `Cannot render a function with params of length [${def.params.length}]`;
+  }
+
+  // First, make sure the first parameter (that is the range vector) is translated if the user selected 'auto'
+  let rangeVector = (model.params ?? [])[0] ?? 'auto';
+
+  if (rangeVector === 'auto') {
+    rangeVector = '$__rate_interval';
+  }
+
+  // Next frame the remaining parameters, but get rid of the first one because it's used to move the
+  // instant vector into a range vector.
+  const params = renderParams(
+    {
+      ...model,
+      params: model.params.slice(1),
+    },
+    {
+      ...def,
+      params: def.params.slice(1),
+      defaultParams: def.defaultParams.slice(1),
+    },
+    innerExpr
+  );
+
+  const str = model.id + '(';
+
+  // Frame the first paramater as a range vector
```

Comment:
Nice, but I guess then this should be `Frame the first or last paramater as a range vector` right?